warc_metadata_sidecar.py

```diff
@@ -0,0 +1,207 @@
+import argparse
+import io
+import logging
+import os
+import re
+import time
+from datetime import timedelta
+
+import chardet
+import pycld2 as cld2
+from fido.fido import Fido
+from warcio.archiveiterator import ArchiveIterator
+from warcio.recordloader import ArcWarcRecord
+from warcio.warcwriter import WARCWriter
+
+
+class ExtendFido(Fido):
+    """A class that extends Fido to override some methods."""
+    def blocking_read(self, file, bytes_to_read):
+        bytes_read = 0
+        buffer = b''
+        while bytes_read < bytes_to_read:
+            readbuffer = file.read(bytes_to_read - bytes_read)
+            buffer += readbuffer
+            bytes_read = len(buffer)
+            # break out if EOF is reached.
+            if not readbuffer:  # this also works with if readbuffer == b'':
+                break
+        return buffer
+
+    def print_matches(self, fullname, matches, delta_t, matchtype=''):
```

Comment:
It would be helpful if there is a docstring for this method and why we are overriding it.